src/Adapters/Upchuck.php

```diff
@@ -47,9 +47,10 @@ public function __construct(Helpers $helpers,
 	 * Duplicate a file given it's URL
 	 *
 	 * @param  string $url
+     * @param  Illuminate\Database\Eloquent\Model $clone
 	 * @return string
 	 */
-	public function duplicate($url) {
+	public function duplicate($url, $clone) {
```

Comment:
What if you give this an explicit default value, even `$clone = null`, so it's not breaking? I haven't tried this out, but would that fix the case of someone who had overrid this without the new arg? 

```suggestion
	public function duplicate($url, $clone = null) {
```

Reply:
Even if you make it nullable it would still be a breaking change, because the interface has changed. If you have created your own adapter, you will have to update it with the extra argument (whether you use it or not) to conform to the interface contract.
Seeing that the duplicate method is only called from one specific place in the cloner class (and it's not likely it will be called anywhere else) where the clone is always available, I don't see the benefit of making making the argument nullable.

Reply:
Cool sounds good.